### PR TITLE
Fix gcc9 warning in HeavyFlavorAnalysis/RecoDecay

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

Fixes a warning in HeavyFlavorAnalysis/RecoDecay

#### PR validation:

Builds without warning

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: HeavyFlavorAnalysis/RecoDecay/src/BPHPlusMinusCandidate.cc [hunk 1]
```diff
@@ -73,6 +73,7 @@ void BPHPlusMinusCandidate::add(
                                           << "already containing same sign particle, add rejected";
         return;
       }
+      [[fallthrough]];
```
Comment (fabiocos) [inline HeavyFlavorAnalysis/RecoDecay/src/BPHPlusMinusCandidate.cc]: @santocch @mrodozov would not a break be more correct? The cases look mutually exclusive to me

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

File: HeavyFlavorAnalysis/RecoDecay/src/BPHPlusMinusCandidate.cc (hunk 1 above)
Comment (santocch) [reply HeavyFlavorAnalysis/RecoDecay/src/BPHPlusMinusCandidate.cc]: - First call: no particles in the collection... so the routine just add the particle regardless of its charge...
   - Second call: check the charge (it has to be the opposite of the first added particle). If it's opposite then OK fallthrough --> add the particle. If it's the same charge then it's not OK and exit from the routine
   - Third call: too many particles --> EXIT

The routine's logic looks ok like it is now...

**Comment by santocch** [comment]
+1

**Pull request merged**